libs/nextjs/src/FronteggProvider.tsx

```diff
@@ -66,47 +111,113 @@ const Connector: FC<ConnectorProps> = (_props) => {
       },
       clientId: props.envClientId,
     }),
-    [ props.envAppUrl, props.envBaseUrl, props.envClientId ]
+    [props.envAppUrl, props.envBaseUrl, props.envClientId, props.contextOptions]
   );
 
+  const authOptions: FronteggAppOptions['authOptions'] = useMemo(() => {
+    const tenantsState = tenants
+      ? {
+          tenantTree: null,
+          subTenants: [],
+          tenants,
+          loading: false,
+          ...props.authOptions?.tenantsState,
+        }
+      : undefined;
+    const userData = user
+      ? {
+          ...user,
+          accessToken: accessToken ?? '',
+          refreshToken: refreshToken ?? undefined,
+          ...props.authOptions?.user,
+        }
+      : null;
+    return {
+      ...props.authOptions,
+      onRedirectTo,
+      isLoading: false,
+      isAuthenticated: !!session,
+      hostedLoginBox: hostedLoginBox ?? false,
+      disableSilentRefresh: props.authOptions?.disableSilentRefresh ?? false,
+      user: userData,
+      tenantsState,
+    };
+  }, [
+    accessToken,
+    hostedLoginBox,
+    onRedirectTo,
+    props.authOptions,
+    refreshToken,
+    session,
+    tenants,
+    user,
+  ]);
+
+  const sharedStore = useMemo(
+    () =>
+      createFronteggStore(
+        { context: contextOptions },
+        storeHolder.current,
+        props.previewMode,
+        authOptions,
+        {
+          auth: authOptions ?? {},
+          audits: props.auditsOptions ?? {},
+        },
+        false,
+        props.urlStrategy
+      ),
+    [
+      authOptions,
+      contextOptions,
+      props.auditsOptions,
+      props.previewMode,
+      props.urlStrategy,
+    ]
+  );
 
   const app = useMemo(() => {
     let createdApp;
     try {
       createdApp = AppHolder.getInstance(appName ?? 'default');
+      createdApp.store = sharedStore;
     } catch (e) {
       createdApp = initialize(
         {
           ...props,
+          store: sharedStore,
           hostedLoginBox: hostedLoginBox ?? false,
           customLoginBox: customLoginBox ?? false,
           basename: props.basename ?? baseName,
-          authOptions: {
-            ...props.authOptions,
-            onRedirectTo,
-          },
-          contextOptions: {
-            requestCredentials: 'include',
-            ...props.contextOptions,
-            ...contextOptions,
-          },
+          authOptions,
+          contextOptions,
           onRedirectTo,
         },
         appName ?? 'default'
       );
     }
     return createdApp;
-  }, [ appName, props, hostedLoginBox, baseName, onRedirectTo, contextOptions ]);
+  }, [
+    appName,
+    props,
+    hostedLoginBox,
+    baseName,
+    onRedirectTo,
+    contextOptions,
+    customLoginBox,
+    authOptions,
+    sharedStore,
+  ]);
+
   ContextHolder.setOnRedirectTo(onRedirectTo);
 
+  if (isSSR) {
```

Comment:
What is the purpose of this function call?

Reply:
useEffect is being called only client side, and we want to call requestAuthorizeSSR also on server side